packages/core/src/routes/hook.ts

```diff
@@ -21,60 +24,101 @@ export default function hookRoutes<T extends AuthedRouter>(
     }
   );
 
+  router.get(
+    '/hooks/:id',
+    koaGuard({
+      params: object({ id: string() }),
+      response: Hooks.guard,
+      status: [200, 404],
+    }),
+    async (ctx, next) => {
+      const {
+        params: { id },
+      } = ctx.guard;
+
+      ctx.body = await findHookById(id);
+
+      return next();
+    }
+  );
+
   router.post(
     '/hooks',
-    koaGuard({ body: Hooks.createGuard.omit({ id: true }), response: Hooks.guard, status: 200 }),
+    koaGuard({
+      body: createHookGuard,
+      response: Hooks.guard,
+      status: [201, 400],
+    }),
     async (ctx, next) => {
+      const { event, events, enabled, ...rest } = ctx.guard.body;
+      assertThat(events ?? event, new RequestError({ code: 'hook.missing_events', status: 400 }));
+
       ctx.body = await insertHook({
+        ...rest,
         id: generateStandardId(),
-        ...ctx.guard.body,
+        signingKey: generateStandardId(),
+        events: events ?? [],
+        enabled: enabled ?? true,
+        ...conditional(event && { event }),
       });
 
+      ctx.status = 201;
+
       return next();
     }
   );
 
-  router.get(
+  router.patch(
     '/hooks/:id',
     koaGuard({
-      params: z.object({ id: z.string().min(1) }),
+      params: object({ id: string() }),
+      body: updateHookGuard,
       response: Hooks.guard,
       status: [200, 404],
     }),
     async (ctx, next) => {
       const {
         params: { id },
+        body,
       } = ctx.guard;
 
-      ctx.body = await findHookById(id);
+      const { config: configToUpdate, ...rest } = body;
+
+      const hook = await findHookById(id);
+      const { config } = hook;
+
+      ctx.body = await updateHookById(id, {
+        ...rest,
+        config: { ...config, ...configToUpdate },
```

Comment:
This is a workaround for the config type infer.

Since the `updateHookById` function comes from `buildUpdateWhereWithPool` util, it requires the data type to be `Partial<Hook>`, which causes the `config` data to become `{ config: HookConfig | undefined }`, and actually, what we want is to make the data type to be `DeepPartial<Hook>`.

Maybe a better solution is not to allow the user to update config data partially (using the `replace` jsonb mode), which is a common practice in our project.